src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/jvm/old_gen/LevelOneActionBuilder.java

```diff
@@ -0,0 +1,100 @@
+ *
+ * <p>if old gen usage(after full gc) is between 60% - 75%, it has not yet reach the 75% threshold in GC.
+ * So we will downsize each cache by regular step size and leave the queues untouched. By default Actions in this
+ * bucket will ignore priority settings of caches and downsize both caches simultaneously until the capacity
+ * of those caches reaches the lower bounds.
+ *
+ * <p>For field data cache, the lower bound in this bucket is 10% of the heap
+ * and for shard request cache / query cache, it will be 2% of the heap(default ES settings). This is
+ * to free up excessive heap used by fielddata cache or query cache because JVM decider favors stability
+ * over performance.
+ */
+public class LevelOneActionBuilder extends BaseActionBuilder {
+
+  private LevelOneActionBuilder(final NodeKey esNode, final NodeConfigCache nodeConfigCache) {
+    super(esNode, nodeConfigCache);
+  }
+
+  public static LevelOneActionBuilder newBuilder(final NodeKey esNode, final NodeConfigCache nodeConfigCache) {
+    return new LevelOneActionBuilder(esNode, nodeConfigCache);
+  }
+
+  private void addFieldDataCacheAction() {
+    Long capacity = NodeConfigCacheReaderUtil
+        .readCacheMaxSizeInBytes(nodeConfigCache, esNode, ResourceEnum.FIELD_DATA_CACHE);
+    if (capacity == null) {
+      return;
+    }
+    ModifyCacheMaxSizeAction action = new ModifyCacheMaxSizeAction(esNode, ResourceEnum.FIELD_DATA_CACHE, nodeConfigCache,
+        capacity, false, LEVEL_ONE_CONST.CACHE_ACTION_STEP_COUNT);
+    if (action.isActionable()
+        && action.getDesiredCapacityInPercent() >= LEVEL_ONE_CONST.FIELD_DATA_CACHE_LOWER_BOUND) {
+      cacheActionMap.put(ResourceEnum.FIELD_DATA_CACHE, action);
+    }
+  }
+
+  private void addShardRequestCacheAction() {
+    Long capacity = NodeConfigCacheReaderUtil
+        .readCacheMaxSizeInBytes(nodeConfigCache, esNode, ResourceEnum.SHARD_REQUEST_CACHE);
+    if (capacity == null) {
+      return;
+    }
+    ModifyCacheMaxSizeAction action = new ModifyCacheMaxSizeAction(esNode, ResourceEnum.SHARD_REQUEST_CACHE, nodeConfigCache,
+        capacity, false, LEVEL_ONE_CONST.CACHE_ACTION_STEP_COUNT);
+    if (action.isActionable()
+        && action.getDesiredCapacityInPercent() >= LEVEL_ONE_CONST.SHARD_REQUEST_CACHE_LOWER_BOUND) {
+      cacheActionMap.put(ResourceEnum.SHARD_REQUEST_CACHE, action);
+    }
+  }
+
+  @Override
+  protected void registerActions() {
+    addFieldDataCacheAction();
+    addShardRequestCacheAction();
+  }
+
+  /**
+   * generate final action list based on action priority.
+   * The default priority in this level is to downsize both caches simultaneously
+   * unless explicitly overridden by customer yml.
+   * @return final action list based on action priority
+   */
+  // TODO : read priority from yml if customer wants to override default ordering
+  @Override
+  protected void actionPriorityFilter() {
+    actionFilter.put(ResourceEnum.FIELD_DATA_CACHE, true);
```

Comment:
How does this ensure a priority order? Keys in the `HashMap` are not ordered, right? I also didn't understand the significance of `boolean` value field

Reply:
IIUC, you are using the `actionFilter` to check which actions to pick based on different levels? e.g. only the actions allowed at that level will be marked `true`. Is this understanding correct? 
If so, then why don't we just create only the relevant actions in each level and do away with this action filter.

Reply:
The intention of creating an action filter is to de-couple the action generation part away from the priority ordering in conf file. By doing this we can avoid a very large function which adds actions and re-order actions based on rca.conf all together in one place.

Reply:
I see. But from current changes, I don't see any action being added to `actionFilter` as `false`, so it seems like we just add actions to the `actionFilter` map as true, and pick all of them in `build` for each `level*ActionBuilder`. What am i missing?